java/ql/src/Telemetry/ExternalAPI.qll

```diff
@@ -0,0 +1,52 @@
+/** Provides classes and predicates related to handling APIs from external libraries. */
+
+private import java
+private import APIUsage
+private import semmle.code.java.dataflow.ExternalFlow
+
+/**
+ * An external API from either the Java Standard Library or a 3rd party library.
+ */
+class ExternalAPI extends Callable {
+  ExternalAPI() { not this.fromSource() }
```

Comment:
Maybe also exclude methods that are overridden in the source?

Reply:
I'm not quite sure about this one. While yes, we don't actually call the external method in that scenario, it would still be interesting to know what this is used as an SPI and we should look into supporting the real implementation. Or do you see any harm in collecting a bit more data because a codebase decided to override a specific implementation?